app/services/resync_service.rb

```diff
@@ -0,0 +1,86 @@
+# frozen_string_literal: true
+
+class ResyncService < ApplicationService
+  attr_reader :document
+
+  def initialize(document)
+    @document = document
+  end
+
+  def call
+    update_live_edition(document.live_edition) if document.live_edition
+    update_current_edition(document.current_edition) unless
+      document.current_edition == document.live_edition
+  end
+
+private
+
+  def update_live_edition(edition)
+    edition.update!(
+      revision_synced: false,
+      system_political: PoliticalEditionIdentifier.new(edition).political?,
+      government_id: government_id(edition, document),
+    )
```

Comment:
since we are mutating an edition (potentially multiple times) we should lock them so they're not vulnerable to any concurrent changes - we'll also have to run the whole thing within a transaction.